apps/framework-cli/src/framework/core/primitive_map.rs

```diff
@@ -12,8 +15,8 @@ use crate::{
 #[derive(Debug)]
 pub struct PrimitiveMap {
     // This probably should not be a top level item and should be nested under the datamodels
-    enums: Vec<DataEnum>,
-    datamodels: HashMap<PathBuf, DataModel>,
+    enums: HashSet<DataEnum>,
+    datamodels: HashMap<PathBuf, Vec<DataModel>>,
```

Comment:
I think it should clarify in a comment whether this is relative path or absolute path.
If we're storing this object in the DB, then relative path makes more sense as it'll be safe from moving the repo.

But then I don't see the benefit of using the file as key instead of the name of the data model.

Reply:
Ha good comments, I was thinking about the diffs and how to be able to find data models quickly that needed update or deletion. The watcher is giving events by the file.

Reply:
Storing that as relative paths from the root of the project makes sense